_rules/document-has-headings-for-non-repeated-content-047fe0.md

```diff
@@ -0,0 +1,448 @@
+  assets:
+    - _The Three Kingdoms_ by Luo Guanzhong, translation by Yu Sumei (Tuttle publishing, 2014, ISBN 9780804843935)
+    - Image from a Ming Dynasty edition of the Romance of the Three Kingdoms, original kept in the library holdings of Peking University.
+---
+
+## Applicability
+
+This rule applies to any [HTML web page][].
+
+## Expectations
+
+In each test target, there exists an element containing [non-repeated content after repeated content][] for which all the following are true:
+
+- the element has a [semantic role][] of `heading`; and
+- the element is [visible][]; and
+- the element is [included in the accessibility tree][].
+
+## Assumptions
+
+- This rule assumes that headings used to pass [Technique H69: Providing heading elements at the beginning of each section of content][tech h69] have to be [included in the accessibility tree][] in order to be beneficial to users of assistive technologies.
+- This rule assumes that the first non-repeated content is starting a new section of content. If this is not the case, it is possible to fail the rule while still passing [Technique H69: Providing heading elements at the beginning of each section of content][tech h69].
+
+## Accessibility Support
+
+- Having a heading for the non-repeated content is sufficient to pass [Success Criterion 2.4.1: Bypass blocks][sc241]. However, if headings are used for that goal, they will only benefit users who can actually navigate from heading to heading (such a functionality can be provided by browsers, browsers plugins, screen readers or other assistive technologies). Users without any possibility for headings navigation will be left without way of bypassing blocks of repeated content and will still experience accessibility issues. Therefore, it is recommended to provide other ways of bypassing blocks.
+- When headings are rendered without sufficient visual cues, they are not perceived as headings by sighted users. In this case, passing this rule might still fail [Technique H69: Providing heading elements at the beginning of each section of content][tech h69] and [Success Criterion 2.4.1: Bypass blocks][sc241]. Additionally, this is likely a failure of [Success Criterion 1.3.1: Info and Relationships][sc131].
+
+## Background
+
+The intention of this rule is that the heading is at (or near) the start of the main area of content of a document. However, defining the main area of content in a non-ambiguous way is not really doable. Therefore, the rule takes a more lenient position and only requires the heading to be some non-repeated content. Additional conditions on this heading were considered and rejected when writing the rule since it might be acceptable, for example, to have non-repeated content such as breadcrumb before any heading. Therefore, it is possible to pass this rule but still fail [H69: Providing heading elements at the beginning of each section of content][tech h69] and violate [Success Criterion 2.4.1: Bypass blocks][sc241].
```

Comment:
I don't think this section is relevant to the rule. Someone who wants to know the history of this rule can find it in this pull request, we don't need to put that in the rule.

Reply:
I agree with @Jym77 on this. Providing a summary of the discussions that have been going on in the background is a great asset both to users who haven't been involved in the development of the rule and also to our future selves. We frequently have pull requests with 100+ of comments, all scattered throughout various files, reviews, and whatnot. It's a nightmare having to go dig through, which I myself have had to do on _several_ occasions.